packages/frontend/src/providers/ExplorerProvider.tsx

```diff
@@ -667,9 +667,6 @@ function reducer(
                 ...state.unsavedChartVersion.metricQuery.tableCalculations.map(
                     (tc) => tc.name,
                 ),
-                ...(state.unsavedChartVersion.metricQuery.customDimensions?.map(
```

Comment:
We can remove the customDimension from columnOrder logic now.
```
case ActionType.TOGGLE_DIMENSION -> columnOrder
case ActionType.TOGGLE_METRIC -> columnOrder
case ActionType.ADD_CUSTOM_DIMENSION -> columnOrder
case ActionType.SET_COLUMN_ORDER: -> columnOrder
case ActionType.ADD_TABLE_CALCULATION: -> columnOrder
case ActionType.DELETE_TABLE_CALCULATION: -> columnOrder
```

We need to handle update/delete of a custom dimension. At the moment, if I edit a custom dim name it get unselected. 
```
case ActionType.EDIT_CUSTOM_DIMENSION -> replace id in `dimensions` 
case ActionType.REMOVE_CUSTOM_DIMENSION -> remove id in `dimensions` ?
```